feat: Added option to get JWT from 'access_token' query parameter from original url header value

AlexGoris-KasparSolutions opened the pull request:
I tried using this solution but ran into an issue for some edge cases where we need to validate a JWT passed as `access_token` query parameter.
I messed around a bit with the `nginx.ingress.kubernetes.io/auth-snippet` and `nginx.ingress.kubernetes.io/auth-proxy-set-headers` annotations of the nginx-ingress controller, but it seems accessing the query parameters is not possible in those scenarios. I dug into it, it's nginx-ingress creates an intermediate internal `location{}` directive for the external auth service which causes the original request context to get lost.

So the next best thing I could think of was to make this service aware of such a query parameter and get if from the `x-original-url` header that nginx-ingress passes to the auth service.

Let me know if you have feedback or issues you'd like me to resolve.

From a security perspective, passing JWT's in query parameters is not a good practice, since they have high chance to be logged in various places, but alas in some scenarios (websockets, mobile apps which open urls in external browsers, ...) this is AFAIK the only solution.

IvanJosipovic commented:
Hey, thank you for PR, and I think makes sense to me.

Could please write a test or two for this? I would the /auth endpoint to have 100% code coverage.

AlexGoris-KasparSolutions commented:
> Hey, thank you for PR, and I think makes sense to me.
> 
> Could please write a test or two for this? I want the /auth endpoint to have 100% code coverage.

Sorry for taking so long, I've added some tests around this.
I also added your docker username to the image name in the tests where the container is created, since it didn't work for me like it was, I assume because you're logged in with your username docker automatically prefixes it when not specified.
I also couldn't get all tests to pass since I don't have your client secret, I'll open an issue to refactor these tests so we can either test using oauth credentiallsl which can be made public, or by making it easier to switch to your own oauth provider

IvanJosipovic commented:
@AlexGoris-KasparSolutions, its released not, thank you and apologies for the delay.
